<?xml version="1.0"?>
<launch>

  <!-- ========== default settings =========== -->  
  <arg name="world_file" default="$(find moby_stage)/worlds/multi_robots_intersection.world"/>
  <arg name="name0" default="car0" />
  <arg name="name1" default="car1" />
  <arg name="cmap_size_default" default="45" />
  <arg name="car0_init_vel" default="0.1" />
  <arg name="car1_init_vel" default="0.1" />

  <param name="/use_sim_time" value="true"/>

  <!--  ================ Stage ================  -->
  <node pkg="stage_ros" type="stageros" name="stageros" args="$(arg world_file)">
    <param name="base_watchdog_timeout" value="0.5"/>
    <remap from="/robot_0/cmd_vel" to="/$(arg name0)/cmd_vel"/>
    <remap from="/robot_1/cmd_vel" to="/$(arg name1)/cmd_vel"/>
    <remap from="/robot_0/odom" to="$(arg name0)/odom"/>
    <remap from="/robot_1/odom" to="$(arg name1)/odom"/>
    <remap from="/robot_0/base_pose_ground_truth" to="/$(arg name0)/base_pose_ground_truth"/>
    <remap from="/robot_1/base_pose_ground_truth" to="/$(arg name1)/base_pose_ground_truth"/>
  </node>

  <!-- ============= navigation of cars ============  
  <node name="car0_nav" type="car0_linear_nav.py" pkg="moby_stage" launch-prefix="xterm -e python -m pdb">
-->
      
  <node name="car0_nav" type="car0_linear_nav.py" pkg="moby_stage" output="screen">

    <param name="obs_name" type="string" value="$(arg name0)" />
    <param name="init_vel" type="double" value="$(arg car0_init_vel)" />
    <param name="cmap_size" type="double" value="$(arg cmap_size_default)" />
  </node>
  <node name="car1_nav" type="car1_linear_nav.py" pkg="moby_stage"  >
    <param name="obs_name" type="string" value="$(arg name1)" />
    <param name="init_vel" type="double" value="$(arg car1_init_vel)" />
    <param name="cmap_size" type="double" value="$(arg cmap_size_default)" />
  </node>

  <!-- ==== costmap visualization ==== -->
  <node name="car0_costmap_plot" type="car0_costmap_plot.py" pkg="moby_stage" >
    <param name="cmap_size" type="double" value="$(arg cmap_size_default)" />
  </node>

  <node name="car1_costmap_plot" type="car1_costmap_plot.py" pkg="moby_stage" >
    <param name="cmap_size" type="double" value="$(arg cmap_size_default)" />
  </node>

</launch>
